Database.md

```diff
@@ -73,87 +73,141 @@ WHERE
-    	STRING hashsum "UNIQUE NOT NULL"
-    	BOOL   ready   "DEFAULT FALSE NOT NULL"
-    }
-    
-    locations {
-    	UUID   uuid         "PRIMARY KEY; NOT NULL"
-    	UUID   parent_uuid  "INDEX; DEFAULT NULL; FOREIGN KEY directories . uuid"
-    	UUID   owner_uuid   "INDEX; NOT NULL"
-    	STRING name         "INDEX; NOT NULL; UNIQUE unique_directory (parent_uuid, owner_uuid, name)"
-    	UUID   file_uuid    "FOREIGN KEY files . uuid"
-    }
+	files {
+		UUID    uuid                "PRIMARY KEY; NOT NULL"
+		UUID    owner_id            "INDEX; NOT NULL"
+		STRING  name                "INDEX; NOT NULL"
+		UINT    size                "NOT NULL"
+		STRING  hashsum             "UNIQUE; NOT NULL"
+		UUID    main_location_id    "DEFAULT NULL; FOREIGN KEY directories.uuid"
+		UUID    replica_location_id "DEFAULT NULL; FOREIGN KEY directories.uuid"
+		BOOL    ready               "DEFAULT FALSE; NOT NULL"
+	}
+
+	directories {
+		UUID        uuid        "PRIMARY KEY; NOT NULL"
+		UUID        owner_id    "INDEX; NOT NULL"
+		STRING      volume      "NOT NULL"
+		UUID        parent_id   "INDEX; DEFAULT NULL; FOREIGN KEY directories.uuid"
+		STRING      name        "NOT NULL"
+	}
 ```
 
 ### Relations
 
 ```mermaid
 erDiagram
-	files     ||--o{ locations: "Has many"
+	files     ||--o{ directories: "Has many"
 ```
 
 ### Queries
 
 #### Create directory
 
 ```sql
-INSERT INTO locations (uuid, parent_uuid, owner_uuid, name)
+INSERT INTO directories (owner_id, volume, parent_id, name)
 VALUES (?, ?, ?, ?)
 ```
 
 #### Create file
 
 ```sql
-INSERT OR IGNORE INTO files (uuid, size, hashsum, ready)
-VALUES (?, ?, ?, FALSE)
+INSERT OR IGNORE INTO files(owner_id, name, size, hashsum)
+VALUES(?, ?, ?, ?)
+```
+
+#### Make ready 
+
+First, create a new location (directory) for the file (as needed). Note that this operation should be performed twice to create the main and replica locations.
+
+```sql
+INSERT OR IGNORE INTO directories(owner_id, volume, parent_id, name)
+VALUES(?, ?, ?, ?)
+```
+
+Then, update the file with the new locations and set it as ready.
+
+```sql
+UPDATE files
+SET
+	main_location_id    = ?
+	replica_location_id = ?
+	ready               = TRUE
+WHERE
+	uuid                = ?
+```
 
--- - Then create the location
+#### Delete file
 
-INSERT INTO locations (uuid, parent_uuid, owner_uuid, name, file_uuid)
-VALUES (?, ?, ?, ?, ?)
+```sql
+DELETE FROM files
+WHERE
+	owner_id	 = ?
+	AND uuid   = ?
 ```
 
-#### Delete file / directory
+#### Delete directory
+
+First, delete all files in the directory.
 
 ```sql
 DELETE FROM files
 WHERE
-	owner_uuid = ?
+	owner_id	 							= ?
+	AND main_location_id    = ?
+```
+
+Then, delete all subdirectories.
+
+```sql
+DELETE FROM directories
+WHERE
+	owner_id	 = ?
+	AND parent_id = ?
+```
+
+Finally, delete the directory itself.
+
+```sql
+DELETE FROM directories
+WHERE
+	owner_id	 = ?
 	AND uuid   = ?
 ```
 
+**TODO:** Should we also delete all the existing replicas of the files in the directory?
```

Comment:
You can remove this since during file deletion the responsibility is on `worker's` shoulders. But just to make clear, yes our system doesn't have side effects, if someone deletes their files, their will be deleted even from the replicas. Just make sure that when writing the OpenAPI spec you return the entire row so the `Worker` can make the decision properly.